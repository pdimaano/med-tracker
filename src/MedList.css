.card {
    background-color: #5cbee5;
    color: #000000;
    border: 10px solid #5cbee5;
    box-shadow: inset 0 0 0 2px #7d8080;
    box-sizing: border-box;
}

.list-group a {
    margin-top: 1rem;
    text-decoration: none;
    color: #000000;
}

.list-group a:hover {
    text-decoration: none;
    border-left-style: solid;
    border-left-width: 3px;
    border-left-color: #f4f1f1;
}

.list-group a:hover li {
    margin-left: 2px;
}

.list-group-item {
    background-color: transparent;
    border-color: #7d8080;
    font-weight: bold;
    color: #000000;
}

.list-group-item:hover {
    cursor: pointer;
}

.inner-border {
    border: 20px solid #000000;
    box-shadow: inset 0 0 0 10px #000000;
    box-sizing: border-box;
}