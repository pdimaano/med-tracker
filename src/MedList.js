import React from "react";
import { Link } from "react-router-dom";
import "./MedList.css";
import {
  Card,
  CardBody,
  CardTitle,
  ListGroup,
  ListGroupItem,
} from "reactstrap";

/** Show list of medications
 *
 *  Props:
 *  - meds: list of med data objects
 *  - day: day of the week
 *
 */

function MedList({ day, meds }) {
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="fw-bold text-center">{day} Meds</CardTitle>
          <ListGroup>
            {meds.map((med) => (
              <Link to={`/${day.toLowerCase()}/${med.id}`} key={med.id}>
                <ListGroupItem>{med.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default MedList;
