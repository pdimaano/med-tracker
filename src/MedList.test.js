import { screen, fireEvent, render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MedList from "./MedList";

it("renders without crashing for monday meds", async function () {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/monday"]}>
      <MedList day="monday" meds={[{ id: 0, name: "monday med" }]} />
    </MemoryRouter>
  );
  screen.getByText("monday med");
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing for friday meds", async function () {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/friday"]}>
      <MedList day="friday" meds={[{ id: 0, name: "friday med" }]} />
    </MemoryRouter>
  );
  screen.getByText("friday med");
  expect(asFragment()).toMatchSnapshot();
});