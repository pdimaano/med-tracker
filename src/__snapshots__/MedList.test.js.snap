// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing for friday meds 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="fw-bold text-center card-title"
        >
          friday Meds
        </div>
        <ul
          class="list-group"
        >
          <a
            href="/friday/0"
          >
            <li
              class="list-group-item"
            >
              friday med
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders without crashing for monday meds 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="fw-bold text-center card-title"
        >
          monday Meds
        </div>
        <ul
          class="list-group"
        >
          <a
            href="/monday/0"
          >
            <li
              class="list-group-item"
            >
              monday med
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
